### PR TITLE
add Boolean kind to Prim.Boolean

### PR DESCRIPTION
**Pull request by justinwoo**
Adds Boolean kind and True/False types of Boolean to Prim.Boolean. Related to #3383 

Corresponding typelevel-prelude change: https://github.com/justinwoo/purescript-typelevel-prelude/commit/b82137f42bdc0f880ddc36c3986b337d42a42d73

### CODE REVIEW

**Comment by justinwoo** [comment]
Builds timed out. Could someone re-run them if that's the issue?

File: src/Language/PureScript/Constants.hs [hunk 1]
```diff
@@ -437,6 +448,12 @@ pattern SymbolAppend = Qualified (Just PrimSymbol) (ProperName "Append")
 pattern SymbolCons :: Qualified (ProperName 'ClassName)
 pattern SymbolCons = Qualified (Just PrimSymbol) (ProperName "Cons")
 
+pattern SymbolContains :: Qualified (ProperName 'ClassName)
+pattern SymbolContains = Qualified (Just PrimSymbol) (ProperName "Contains")
+
+pattern SymbolBreakOn :: Qualified (ProperName 'ClassName)
+pattern SymbolBreakOn = Qualified (Just PrimSymbol) (ProperName "BreakOn")
```
Comment (LiamGoodacre) [inline src/Language/PureScript/Constants.hs]: Did you mean to add these in this PR?

**Comment by LiamGoodacre** [comment]
I restarted the timed out job.

File: src/Language/PureScript/Constants.hs (hunk 1 above)
Comment (justinwoo) [reply src/Language/PureScript/Constants.hs]: oh, sorry, applied a patch but it was wrong :sweat_smile:

**Comment by justinwoo** [comment]
4/5 ran within time this time through...

**Comment by joneshf** [comment]
If you need to restart the build and don't have access to it on Travis, you can close this PR and open it again. It will restart the build.

I think it means all builds will run again, but it beats waiting for someone else to come along.

**Comment by hdgarrood** [comment]
No, please don’t do that - instead push an empty commit (as the CI log should suggest). As far as I’m aware that’s the only way to get travis to reuse the progress it made during its previous attempt.

**Comment by garyb** [comment]
I've restarted the failing job a couple of times fwiw 😕

**Comment by hdgarrood** [comment]
Restarting jobs which have timed out is generally not very effective in my experience; we should be encouraging the submitter to push empty commits instead.

**Comment by LiamGoodacre** [comment]
(Instead of empty commits you can regenerate your existing commit with `git commit --amend --no-edit` - this new commit will have a different commit hash.  Then force push.)

**Comment by joneshf** [comment]
An alternate to any of our suggestions could be focusing on making CI more reliable 😄.

**Comment by joneshf** [comment]
Looks like we build the haddocks for dependencies of every job in the matrix whether it has `BUILD_TYPE=haddock` or not. The job that's failing doesn't have `BUILD_TYPE=haddock`. Given that it's timing out when uploading the cache–almost done with the build–that extra time with the haddocks is probably enough to push it over the limit. Should we change the jobs to only build haddocks when necessary?

Is it this line? https://github.com/purescript/purescript/blob/f8ca834ebe3d8e5d484e5b55b5d1b8b6f0a783a2/travis/build.sh#L8-L10

I'm more than willing to submit a PR for it.

**Comment by nwolverson** [comment]
@joneshf I applied that suggestion in my fork here https://github.com/purerl/purescript/commit/5b656c8d945b86c905f3a555d221f60a4dc8328b - not quite sure if it helped or I just had stuff cached by this point, but I think by my understanding this should be right - I believe the cache will be separate for the different entries in our build matrix

**Comment by hdgarrood** [comment]
Sorry for the delay in responding. Good spot - yes I think that would help.

**Comment by justinwoo** [comment]
Heyy, this passed now :)

**Comment by LiamGoodacre** [comment]
Some of the BreakOn and Contains stuff is still in this PR, @justinwoo ?

**Comment by justinwoo** [comment]
Finally removed it

**Pull request merged**